webextension/background/auth.js

```diff
@@ -9,7 +9,13 @@ window.auth = (function () {
   let authHeader = null;
   let sentryPublicDSN = null;
 
-  chrome.storage.local.get(["registrationInfo"], (result) => {
+  chrome.storage.local.get(["registrationInfo"], catcher.watchFunction((result) => {
+    if (chrome.runtime.lastError) {
```

Comment:
Would you mind changing this to use a Promise instead, maybe wrap `storage.local.get` in a `catcher.watchPromise` call?

According to the [`runtime.lastError` docs](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/runtime/lastError), you don't need to use `lastError` if you use the Promise-based versions of APIs.

It seems like pageshot uses promises pretty extensively, so it'd be nice to make these spots consistent while we're in here.